script.js

```diff
@@ -1,4 +1,198 @@
+function drawCard(deck) {
+  return deck.shift();
+}
+
+// Helper function to display the hands of the players
+function displayHands(hands) {
+  if (gameState === "player" && playerScore < 22) {
+    return `Player's hand:<br> ${cardsToString(
+      hands.player
+    )}<br>Dealer's hand: ${hands.dealer[0].rank}, [Hidden Card]`;
+  } else {
+    return `Player's hand:<br> ${cardsToString(
+      hands.player
+    )}<br>Dealer's hand:<br> ${cardsToString(hands.dealer)}`;
+  }
+}
+
+// Helper function to convert a single card array to a string representation
+function oneCardToString(card) {
+  return `${card.rank} of ${card.suit}`;
+}
+
+// Helper function to convert multiple card array to a string representation
+function cardsToString(cards) {
+  return cards.map((card) => oneCardToString(card)).join("<br>");
+}
+
+// Helper function to handle the dealer's turn
+function dealerTurn(hands) {
+  while (true) {
+    dealerScore = calculateScore(hands.dealer);
+    if (dealerScore < 17) {
+      hands.dealer.push(drawCard(deck));
+    } else {
+      playerScore = calculateScore(hands.player);
+      var message = `Player's cards:<br> 
+        ${cardsToString(hands.player)}<br>
+        Dealers's cards:<br> ${cardsToString(hands.dealer)}<br>
+        Player's final score: ${playerScore}<br>
+        Dealer's final score: ${dealerScore}<br>`;
+      gameState = "end";
+
+      if (playerScore > 21 && dealerScore > 21) {
+        message += "Both players busted! It's a tie!";
+        message += `<br><img src="https://media.tenor.com/OiBRj-lhVUgAAAAS/well-call-it-a-tie-lucifer-morningstar.gif" width="477" height="480" class="giphy-embed">`;
+      } else if (playerScore <= 21 && dealerScore <= 21) {
+        if (playerScore > dealerScore) {
+          message += "Player wins!";
+          message += `<br><img src="https://media.tenor.com/FAIxgHJ6oX4AAAAi/white-chicken.gif" width="477" height="480" class="giphy-embed">`;
+        } else if (playerScore < dealerScore) {
+          message += "Dealer wins!";
+          message += `<br><img src="https://media.tenor.com/PUm1wkSXINMAAAAC/teasing-amanda-cee.gif" width="477" height="480" class="giphy-embed">`;
+        } else {
+          message += "It's a tie!";
+          message += `<br><img src="https://media.tenor.com/OiBRj-lhVUgAAAAS/well-call-it-a-tie-lucifer-morningstar.gif" width="477" height="480" class="giphy-embed">`;
+        }
+      } else if (playerScore > 21 && dealerScore <= 21) {
+        message += "Player busted! Dealer wins!";
+        message += `<br><img src="https://media.tenor.com/PUm1wkSXINMAAAAC/teasing-amanda-cee.gif" width="477" height="480" class="giphy-embed">`;
+      } else if (playerScore <= 21 && dealerScore > 21) {
+        message += "Dealer busted! Player wins!";
+        message += `<br><img src="https://media.tenor.com/FAIxgHJ6oX4AAAAi/white-chicken.gif" width="477" height="480" class="giphy-embed">`;
+      }
+
+      return message;
+    }
+  }
+}
```

Comment:
This function is pretty long and loaded, code blocks in some of the `if`/ `else if` statements can be abstracted into its own functions. e.g. `handleBothPlayerAndDealerBurst` or something like that